packages/@aws-cdk/aws-pipes-alpha/lib/source.ts

```diff
@@ -100,3 +103,42 @@ export interface ISource {
    */
   grantRead(grantee: IRole): void;
 }
+
+/**
+ * Source interface with dead-letter target
+ */
+interface ISourceWithDlq extends ISource {
+  grantDlqPush(grantee: IRole, deadLetterTarget?: IQueue | ITopic): void;
+}
+
+export abstract class SourceWithDlq implements ISourceWithDlq {
```

Comment:
You can combine these and remove `ISourceWithDlq`, since `grantDlqPush` is implemented below:
```suggestion
export abstract class SourceWithDlq implements ISource {
```